goldmane/pkg/aggregator/aggregator.go

```diff
@@ -278,102 +303,31 @@ func (a *LogAggregator) Stream(req *proto.FlowStreamRequest) (*Stream, error) {
 
 // List returns a list of flows that match the given request. It uses a channel to
 // synchronously request the flows from the aggregator.
-func (a *LogAggregator) List(req *proto.FlowListRequest) ([]*proto.FlowResult, error) {
+func (a *LogAggregator) List(req *proto.FlowListRequest) ([]*proto.FlowResult, int64, error) {
 	respCh := make(chan *listResponse)
 	defer close(respCh)
 	a.listRequests <- listRequest{respCh, req}
 	resp := <-respCh
-	return resp.flows, resp.err
+	return resp.flows, resp.total, resp.err
 }
 
-func (a *LogAggregator) Hints(req *proto.FilterHintsRequest) ([]*proto.FilterHint, error) {
+func (a *LogAggregator) Hints(req *proto.FilterHintsRequest) (*proto.FilterHintsResponse, error) {
 	logrus.WithField("req", req).Debug("Received hints request")
 
-	// For now, we just convert this to a list request and return the hints based on the results.
-	// This is quick and dirty - we can plumb the implementation down to the Index later if needed.
-	//
-	// We configure the SortBy based on the FilterType, as this is the most likely way that the
-	// hints will be used and it allows us to use the index for the most common case.
-	sortBy := proto.SortBy_Time
-	switch req.Type {
-	case proto.FilterType_FilterTypeDestName:
-		sortBy = proto.SortBy_DestName
-	case proto.FilterType_FilterTypeDestNamespace:
-		sortBy = proto.SortBy_DestNamespace
-	case proto.FilterType_FilterTypeSourceName:
-		sortBy = proto.SortBy_SourceName
-	case proto.FilterType_FilterTypeSourceNamespace:
-		sortBy = proto.SortBy_SourceNamespace
-	}
-
-	// Sanitize the time range, resolving any relative time values.
-	req.StartTimeGte, req.StartTimeLt = a.normalizeTimeRange(req.StartTimeGte, req.StartTimeLt)
+	respCh := make(chan *filterHintsResponse)
+	defer close(respCh)
+	a.filterHintsRequests <- filterHintsRequest{respCh, req}
+	resp := <-respCh
 
-	flows, err := a.List(&proto.FlowListRequest{
-		SortBy:       []*proto.SortOption{{SortBy: sortBy}},
-		Filter:       req.Filter,
-		StartTimeGte: req.StartTimeGte,
-		StartTimeLt:  req.StartTimeLt,
-	})
-	if err != nil {
-		return nil, err
+	var hints []*proto.FilterHint
+	for _, key := range resp.keys {
```

Comment:
should this field be `resp.values`? It's going into the `Value: ` field of the response.